### PR TITLE
[admin] Tgui secrets

### PR DESCRIPTION
**Pull request by swissloaf**
Makes the admin secrets panel TGUI

https://github.com/BeeStation/BeeStation-Hornet/pull/2033 Original PR
image

Looks better than the current version
image

#### Changelog

:cl:  
rscadd: Adds admin secret panel TGUI
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
i cant help but notice that your secrets panel screenshot is outdated but the code looks fine

**Comment by swissloaf** [comment]
> i cant help but notice that your secrets panel screenshot is outdated but the code looks fine

the new one I posted is updated, old one is from the wiki. its about 5-6 months old

**Comment by missatessatessy** [comment]
So much nicer

**Comment by JamieD1** [comment]
@TheGamerdk

**Review by fluffe9911 — APPROVED** [review]
Code is probably fine but gots a conflict

**Review by alexkar598 — CHANGES_REQUESTED** [review]
This shit needs to be documented

File: code/modules/admin/secrets.dm [hunk 1]
```diff
@@ -1,123 +1,119 @@
...
+	if(!admin_datum) // Should have already been blocked by ui_state, but juuust in case
+		message_admins("[usr] sent a request to interact with the secrets panel without sufficient rights.")
+		log_admin_private("[usr] sent a request to interact with the secrets panel without sufficient rights.")
+		return
 	var/datum/round_event/E
 	var/ok = 0
-	switch(item)
+	switch(action)
 		if("admin_log")
-			var/dat = "<HTML><HEAD><meta charset='UTF-8'></HEAD><BODY><B>Admin Log<HR></B>"
```
Comment (TheTrigg) [inline code/modules/admin/secrets.dm:103]: You probably don't want to revert this UTF-8 stuff

**Pull request closed without merging**